numba/tests/test_obj_lifetime.py

```diff
@@ -248,8 +248,10 @@ class TestObjLifetime(TestCase):
     """
 
     def compile(self, pyfunc):
-        cr = compile_isolated(pyfunc, (types.pyobject,), flags=forceobj_flags)
-        return cr.entry_point
+        # Note: looplift must be disabled. The test require the function
+        #       control-flow to be unchanged.
+        cfunc = jit((types.pyobject,), forceobj=True, looplift=False)(pyfunc)
```

Comment:
```suggestion
        cfunc = jit((types.pyobject,), nopython=False, forceobj=True,
                    looplift=False)(pyfunc)
```
else it raises a warning.